tools/autograd/gen_python_functions.py

```diff
@@ -1541,3 +886,115 @@ def op_name(declaration):
                 '{}: name ends with \'_out\', expecting output params'.
                 format(declaration['name']))
         return name
+
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ #
+#
+#                       Codegen API Integration
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ #
+
+# These helper functions allow us to call the new codegen API from the
+# old codegen script (which operates on Declarations.yaml).
+
+# TODO: remove all these HACKs after migration is completed!
+
+# function schema str -> NativeFunction
+NF_TABLE: Optional[Dict[str, NativeFunction]] = None
+
+def init(native_yaml_path: str) -> None:
+    from tools.codegen.gen import parse_native_yaml
+    global NF_TABLE
+    NF_TABLE = {str(f.func): f for f in parse_native_yaml(native_yaml_path)}
+
+# Multiple decl entries can map to the same native function (because of deprecated decl).
+def decl_to_native_function(decl: Dict[str, Any]) -> NativeFunction:
+    assert NF_TABLE is not None, 'need to initialize codegen.api.python with init()'
+    function_schema_str = decl['schema_string'][6:]
+    assert function_schema_str in NF_TABLE, f'cannot find func: {function_schema_str}'
+    return NF_TABLE[function_schema_str]
+
+# Each decl entry has unique python signature.
+def decl_to_python_signature(decl: Dict[str, Any], *, method: bool) -> PythonSignature:
+    f = decl_to_native_function(decl)
+
+    @with_native_function
+    def go(f: NativeFunction) -> PythonSignature:
+        return signature(f, method=method)
```

Comment:
This refactor probably would have come in handy here: https://github.com/pytorch/pytorch/pull/45277/files#diff-6382c847dac2d011f822c492e61f9fa0eceb47c2d6fc5fb4f0c42af726c1a3aaR110